examples/function-closure/function_closure.bal

```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
+
+public function main() {
+    string[] names = ["Ana", "Alice", "Bob"];
+
+    // Define a function to modify and return the variable 'names' that declares outside the scope.
+    var innerFunction = function() returns string[] {
```

Comment:
It is better if the function name shows what is doing.

Reply:
Sent a suggestion.